### PR TITLE
Chats creations

### PR DESCRIPTION
**Pull request by agupirotto**
Endpoint to create new chat
This pull request is for the implementation of the back-end tast, https://trello.com/c/tQ2i567N , for the story card,https://trello.com/c/GRYbvTEh
Poor endpoint documentation, tips are accepted!

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ class User < ActiveRecord::Base
   include DeviseTokenAuth::Concerns::User
 
   validates :role, :first_name, :last_name, :email, presence: true
-  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, uniqueness: { case_sensitive: false }
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, uniqueness: { case_sensitive: true }
```
Comment (Phoyu) [inline app/models/user.rb]: The URI::MailTo::EMAIL_REGEXP regex is case insensitive, so example@mail.com is the same as Example@Mail.Com.
case_sensitive must be false

File: README.md [hunk 2]
```diff
@@ -1 +1,9 @@
-# pis2019-chat-dashboard
\ No newline at end of file
+# API Endpoints: 
+
+### Create a new chat:  
+```
+post app_domain/api/v1/chats
```
Comment (petterboussard) [inline README.md]: You should tell the front-end team if they need to provide any params in order to make the request and what the endpoint response is going to look like.

File: app/controllers/api/v1/chats_controller.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Api::V1::ChatsController < ApplicationController
+  def create
+    new_chat = Chat.new
+    return render json: new_chat, status: :created if new_chat.save
```
Comment (petterboussard) [inline app/controllers/api/v1/chats_controller.rb]: Do they really need the whole chat or just the chat_id?

File: spec/factories/users.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :user do
+    password { Faker::Internet.password }
+    first_name { Faker::Name.first_name }
+    last_name { Faker::Name.last_name }
+    role { Faker::Number.within(range: 1..2) }
+    email { Faker::Internet.email }
+    site { FactoryBot.create(:site) }
```
Comment (petterboussard) [inline spec/factories/users.rb]: You can just type site and the site factory will be called, there is no need for **{ FactoryBot.create(:site) }**

File: README.md [hunk 5]
```diff
@@ -58,4 +58,13 @@ As a response it returns the assigned chat and the messages it contains as a JSO
...
+{ **state**, **id**, **solved**, **messages**: [ { id, body, user_id, chat_id, ... }, ... ], ... } 
+
+
+### Create a new chat:  
+```
+post app_domain/api/v1/chats
+```
+> This endpoint is intended to be used only from the widget.  
+
+**Request body schema**: it doesn't need any parameters to be invoked
```
Comment (petterboussard) [inline README.md]: You should document about what the response object will look like.
Should be something like: { chat_id: id }

File: app/controllers/api/v1/chats_controller.rb [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 class Api::V1::ChatsController < ApplicationController
   before_action :authenticate_user!, only: [:new_dashboard_message]
+  def create
+    new_chat = Chat.new
+    return render json: new_chat, status: :created if new_chat.save
```
Comment (petterboussard) [inline app/controllers/api/v1/chats_controller.rb]: Please just return the new_chat id.
Should be like: 
return render json: { chat_id: new_chat.id }, status: :created if new_chat.save

**Pull request merged**